src/ocean/time/Time.d

```diff
@@ -149,131 +149,166 @@ struct TimeSpan
         `));
 
         /**
-         * Add the TimeSpan given to this TimeSpan returning a new TimeSpan.
+         * Add or subtract the TimeSpan given to this TimeSpan returning a
+         * new TimeSpan.
          *
-         * Params: t = A TimeSpan value to add
-         * Returns: A TimeSpan value that is the sum of this instance and t.
+         * Params:  op = operation to perform
+         *          t = A TimeSpan value to add or subtract
+         * Returns: A TimeSpan value that is the result of this instance, the
+         *          operation, and t.
          */
-        TimeSpan opAdd(TimeSpan t)
+        TimeSpan opBinary (string op) (TimeSpan t) if (op == "+" || op == "-")
         {
-                return TimeSpan(ticks_ + t.ticks_);
+            mixin("return TimeSpan(ticks_ " ~ op ~ " t.ticks_);");
         }
 
-        /**
-         * Add the specified TimeSpan to this TimeSpan, assigning the result
-         * to this instance.
-         *
-         * Params: t = A TimeSpan value to add
-         * Returns: a copy of this instance after adding t.
-         */
-        TimeSpan opAddAssign(TimeSpan t)
+        unittest
         {
-                ticks_ += t.ticks_;
-                return this;
+            TimeSpan time_span;
+            time_span.ticks_ = 5;
+
+            TimeSpan test_span;
+            test_span.ticks_ = 10;
+
+            auto res = time_span + test_span;
+            test!("==")(res.ticks(), 15);
+
+            test_span.ticks_ = 3;
+
+            res = time_span - test_span;
+            test!("==")(res.ticks(), 2);
         }
 
         /**
-         * Subtract the specified TimeSpan from this TimeSpan.
+         * Add or subtract the specified TimeSpan to this TimeSpan, assigning
+         * the result to this instance.
          *
-         * Params: t = A TimeSpan to subtract
-         * Returns: A new timespan which is the difference between this
-         * instance and t
+         * Params:  op = operation to perform
+         *          t = A TimeSpan value to add or subtract
+         * Returns: a copy of this instance after adding or subtracting t.
          */
-        TimeSpan opSub(TimeSpan t)
+
```

Comment:
cosmetic: you might want to remove this extra empty line

Reply:
hmm, I couldn't seem to find this one.

Reply:
I've just mentioned because there were not empty line before and now there is one...anyway let's keep it as it is